packages/components/src/utils/jss.ts

```diff
@@ -145,3 +145,23 @@ export const mergeDeep = <T extends Record<string, any>>(...objects: T[]): T =>
     return prev;
   }, {} as T);
 };
+
+export const deepEqual = <T extends Record<string, any>>(object1: T, object2: T) => {
+  const keys1 = Object.keys(object1);
+  const keys2 = Object.keys(object2);
+
+  if (keys1.length !== keys2.length) {
```

Comment:
Let's improve branching. 
If this would check for `===`, then the for loop could be within the `if` block and then we add a `else`  block with the `return true`.